deploy/role.yaml

```diff
@@ -78,4 +78,13 @@ rules:
   verbs:
   - get
   - list
+  - watch
+- apiGroups:
+  - org.eclipse.che
```

Comment:
This role is referred in RoleBinging for the toolchain operator SA for the namespaces where the operator is deployed. So, if we have it in the role only then we will be able to create CheCluster only in openshift-operators namespace, won't we?
So, we need to keep that permission in the ClusterRole or I'm missing something?

Reply:
no, because we support only AllNamespaces install mode so the permissions will be applied to all namespaces (including toolchain-workspaces)

Reply:
After all, you could say the same thing about Subscriptions as well and we already verified that it's working fine - we were able to manage them in the toolchain-workspaces namespace.

Reply:
OK. I will need to look at how OLM creates roles/rolebindings in all namespaces mode (I guess it creates it in every namespace). But wouldn't it make more sense to keep all permissions we need for all the namespaces in the ClusterRole anyway?
I mean both option should work (in case of all namespaces mode) but instead of forcing OLM propagating the role/rolebinding in all namespaces why not just using the only ClusterRole instance for everything?

Reply:
BTW, Pipelines uses ClusterRole only which IMO makes sense.

Reply:
Yes,both options should work. Originally, we had it in the clusrerole, but @xcoulon asked me to move it if it's not really needed in cluster-role permissions. Anyway, I'm fine with both options so let's discuss it during our sync meeting and agree on one thing together.

Reply:
As discussed, it was me who misunderstood the way we define permissions in the `cluster_role.yaml` vs `role.yaml` files :/ Having a single cluster role (with permissions on namespaced resources), which can be bound to users via a `rolebinding` should avoid creating duplicate roles for each user/namespace in etcd, while easing their maintenance.

Reply:
ok, based on our discussion I created a new PR where I move everything from the role.yaml to cluster_role.yaml https://github.com/codeready-toolchain/toolchain-operator/pull/47